### PR TITLE
Feature/add unitSpace prop in hero

### PR DESCRIPTION
**Pull request by ektasawant**
Fixes [348](https://github.com/pxblue/angular-component-library/issues/348) .

#### Changes proposed in this Pull Request:
- Add unitSpace prop in Hero component

#### To Test:
-  http://localhost:6006/?path=/story/components-hero--with-channelvalue-children
- http://localhost:6006/?path=/story/components-hero--within-a-herobanner

### CODE REVIEW

**Review by huayunh — COMMENTED** [review]
Need to add your changes to docs too

**Comment by ektasawant** [comment]
> changes

Could you please elaborate more? Where I need to update docs?

**Comment by ektasawant** [comment]
> Need to add your changes to docs too

Could you please elaborate more? Where I need to update docs?

**Review by huayunh — APPROVED** [review]
Oh sorry. I was referring to https://github.com/pxblue/angular-component-library/blob/feature/add-unitspace-prop-in-hero/docs/Hero.md. 

Change looks good.

Can you also tweak the full config story's `unitSpace`?
image

So there should be no space between "96" and "/100". The "C" for temperature is not technically correct, maybe `°C` or some other units.

File: components/src/core/hero/hero.component.ts [hunk 1]
```diff
@@ -65,6 +65,8 @@ export class HeroComponent implements OnChanges, AfterViewInit, AfterContentChec
     @Input() units: string;
     /** The value for the channel */
     @Input() value: string;
+    /** Show a space between the value and units for the channel */
+    @Input() unitSpace: string;
```
[the remark this answers is not in the document]
Comment (ektasawant) [reply components/src/core/hero/hero.component.ts]: Done.

**Comment by ektasawant** [comment]
> Oh sorry. I was referring to https://github.com/pxblue/angular-component-library/blob/feature/add-unitspace-prop-in-hero/docs/Hero.md.
> 
> Change looks good.
> 
> Can you also tweak the full config story's `unitSpace`? image
> 
> So there should be no space between "96" and "/100". The "C" for temperature is not technically correct, maybe `°C` or some other units.

Done. 
image

**Comment by huayunh** [comment]
Still need to update your API table. Otherwise looks good.

**Review by emclaug2 — COMMENTED** [review]
Can you also add a `unitSpace` knob onto the "full config" hero story?

**Comment by emclaug2** [comment]
I'm thinking this should be the latest feature branch that goes into release/6.0.0 before we publish.  Ideally we can publish @pxblue/angular-component-library v6.0.0 tomorrow after our planning meeting or on Friday.

**Comment by ektasawant** [comment]
> Whenever this is ready for review I think it's looking pretty good.
> 
> One last thing I noticed is that the type for the `unitSpace` has been defined in the ChannelValue component & can be imported into the HeroComponent so we have these components share the same type. `UnitSpaceType`

Sure let me update it. 
@emclaug2 , could you please let me know do I need below changes in this PR?

Screenshot 2021-09-30 at 7 41 01 PM

File: docs/Hero.md [hunk 2]
```diff
@@ -53,6 +53,7 @@ Parent element (`<pxb-hero>`) attributes:
 | iconSize | The size of the primary icon (10-48)   | `number`             | no       | 'normal' (36) |
 | label    | The text shown below the Channel Value | `string`             | yes      |               |
 | units    | Text to show after the value           | `string`             | no       |               |
+| unitSpace  | Show/Hide spacing between the value and units  | `'auto'` \| `'hide'` \| `'show'` | yes      | 'auto'  |
```
Comment (joebochill) [inline docs/Hero.md:56]: I believe this should be required: _no_ (it has a default value and thus should be optional).

**Comment by ektasawant** [comment]
> Whenever this is ready for review I think it's looking pretty good.
> 
> One last thing I noticed is that the type for the `unitSpace` has been defined in the ChannelValue component & can be imported into the HeroComponent so we have these components share the same type. `UnitSpaceType`

Done.

**Pull request merged**